### DISCUSSION TITLE
Adding buckling lenghts to SAF - how to do it?

### DISCUSSION THREAD

**Discussion by sonkaste17** [StructuralAnalysisFormat/StructuralAnalysisFormat-Doc#97] [Ideas]
We need to investigate options for storing a defining the buckling lengths in SAF.

**Comment by robinblazek** [comment]
Hi all,

we would like to introduce here the proposal for new SAF object, which should include definition of buckling system. This object then could be referenced by (applied to) 1D member. 

In principal by the system we want to define, which supports should be taken into the consideration for each direction (type) of buckling and also which factor should be used for calculation. Factor then can be used for multiplication of the system length of the span for the member. 

New object: **BucklingSystem**

Property | Example | Description
-- | -- | --
Name | BS1 | Unique name of the buckling system
Buckling supports y | Yes; No; Yes | Defines, which supports should be taken into consideration for buckling **around** major axis of the cross-section.
Factor y | 1.3 | Buckling factor for major axis of the cross-section, which is used for multiplication of the system length of the span. 
Buckling supports z | Yes; Yes; Yes | Defines, which supports should be taken into consideration for buckling **around** minor axis of the cross-section. 
Factor z | 1.3;1.8 | Buckling factor for minor axis of the cross-section, which is used for multiplication of the system length of the span. 
Buckling supports y-z | Yes; No; Yes | Defines, which supports should be taken into consideration for torsional (flexural) buckling. 
Factor y-z | 1.3 | Buckling factor for torsional (flexural) buckling, which is used for multiplication of the system length of the span. 
Buckling supports LTB | Yes; Yes; Yes | Defines, which supports should be taken into consideration for lateral-torsional buckling.
Factor LTB | 1.2; 1.6 | Buckling factor for lateral-torsional buckling, which is used for multiplication of the system length of the span.
Id |   | Unique attribute designation

**Note I:** By "supports" are meant all the nodes (begin, all internal and end node) which are part of the 1D member. All those nodes on the 1D member are considered as possible supports and by Yes/No it is defined, if it is valid in the specific direction.

 **Note II:** Properties "Buckling supports X", where "X" is direction, are describing supports, which are taken into consideration, for the buckling **AROUND** the axis "X". So for example "Buckling supports y" is describing behaviour of the buckling **AROUND** axis Y, which means, that supports are actually applied here for the "z" direction. Used directions are **according to the principal axis system**, so the major (Y) and minor (Z) direction of the CSS.  

**Note III:** Supports are dividing the member into multiple spans, which the factors are defined for. Factors have to be defined in the specific direction always for all the spans. Here are some examples to illustrate it: 
**Ex.1:** If "Buckling supports y" is defined as "Yes; No; No; Yes" -> it means, valid supports are only in the start and end point of the 1D member, which means, that there is only 1 span -> there will be only 1 value of the "Factor y". 
**Ex.2:** If "Buckling supports y" is defined as "Yes; No; Yes; No" -> it means, there are 2 valid supports, but this time 2 spans (first one between the supports and second one from the second support to the free end of the member) -> there will be 2 values of the "Factor y". 
**Ex.3:**
<img width="916" alt="image" src="https://user-images.githubusercontent.com/88313526/221873397-09801b97-030e-4b66-b091-63755f9f4647.png">  

As always, feel free to ask questions or suggest changes. Any feedback is welcome!
